ambry-commons/src/test/java/com/github/ambry/compression/BaseCompressionTest.java

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.compression;
+
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.TestUtils;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class BaseCompressionTest {
+
+  @Test
+  public void testGetAlgorithmNameBinary() throws Throwable {
+    // Test: A single-letter algorithm name.
+    BaseCompression compressor = Mockito.mock(BaseCompression.class);
+    Mockito.when(compressor.getAlgorithmName()).thenReturn("A");
+    byte[] binary = (byte[]) MethodUtils.invokeMethod(compressor, true, "getAlgorithmNameBinary");
```

Comment:
what is the purpose of using this indirect method call? we can just do compressor.getAlgorithemNameBinary.